cmd/kwild/server/build.go

```diff
@@ -763,7 +763,7 @@ func buildSnapshotter(d *coreDependencies) *statesync.SnapshotStore {
 	return ss
 }
 
-func buildStatesyncer(d *coreDependencies) *statesync.StateSyncer {
+func buildStatesyncer(d *coreDependencies, closer *closeFuncs) *statesync.StateSyncer {
```

Comment:
I think this can be `func buildStatesyncer(d *coreDependencies, db sql.ReadTxMaker)` and take the `db` from `buildServer`.

I recall that `buildEventStore` need its own instance because it perform writes to the events table outside of the consensus txn, but statesyncer just needs a read tx, so it's fine.